app/screens/browse_channels/channel_list.tsx

```diff
@@ -81,54 +80,66 @@ export default function ChannelList({
 
     const renderLoading = useCallback(() => {
         return (
-            <Loading
-                containerStyle={style.loadingContainer}
-                style={style.loading}
-                color={theme.buttonBg}
-            />
+            <View style={[style.noResultContainer, {paddingBottom: keyboardHeight}]}>
+                <Loading
+                    containerStyle={style.loadingContainer}
+                    style={style.loading}
+                    color={theme.buttonBg}
+                />
+            </View>
         );
 
     //Style is covered by the theme
-    }, [theme]);
+    }, [theme, keyboardHeight]);
 
     const renderNoResults = useCallback(() => {
-        if (isSearch) {
+        if (term) {
             return (
-                <View style={style.noResultContainer}>
-                    <FormattedText
-                        id='mobile.custom_list.no_results'
-                        defaultMessage='No Results'
-                        style={style.noResultText}
-                    />
+                <View style={[style.noResultContainer, {paddingBottom: keyboardHeight}]}>
+                    <NoResultsWithTerm term={term}/>
                 </View>
             );
         }
 
         return (
-            <View style={style.noResultContainer}>
+            <View style={[style.noResultContainer, {paddingBottom: keyboardHeight}]}>
                 <FormattedText
                     id='browse_channels.noMore'
                     defaultMessage='No more channels to join'
                     style={style.noResultText}
                 />
             </View>
         );
-    }, [style, isSearch]);
+    }, [style, term, keyboardHeight]);
 
     const renderSeparator = useCallback(() => (
         <View
             style={style.separator}
         />
     ), [theme]);
 
+    useEffect(() => {
+        const show = Keyboard.addListener('keyboardWillShow', (event) => {
+            setKeyboardHeight(event.endCoordinates.height);
+        });
+
+        const hide = Keyboard.addListener('keyboardWillHide', () => {
+            setKeyboardHeight(0);
+        });
+
+        return () => {
+            show.remove();
+            hide.remove();
+        };
+    }, []);
```

Comment:
this code is duplicate in both screens, I tried to add it to the `no_results_with_term` component, but the component is mounted after the Keyboard has been opened, so the UX is not really what we want. So it is expected to be applied in the components that needs to position the no results component or others based on the keyboard being opened or not.

Reply:
This can be transformed into a function `registerKeyboard(setHeight: (h: number) => void)` and extracted to some `utils` or `common_effects` file.

Another option is to make it a custom hook.

Nevertheless, 0/5 on whether we should go that direction or just keep a bit of repetition.

Reply:
I like the hook idea, will do